bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/FirmwareBuilder.java

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.thing.binding.firmware;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.internal.firmware.FirmwareImpl;
+
+/**
+ * The builder to create a {@link Firmware}.
+ *
+ * @author Thomas Höfer - Initial contribution
+ * @author Dimitar Ivanov - extracted as separate class for Firmware.
+ */
+public final class FirmwareBuilder {
+
+    private final ThingTypeUID thingTypeUID;
+    private final String version;
+    private String vendor;
+    private String model;
+    private boolean modelRestricted;
+    private String description;
+    private String prerequisiteVersion;
+    private String changelog;
+    private URL onlineChangelog;
+    private transient InputStream inputStream;
+    private String md5Hash;
+    private Map<String, String> properties;
+
+    public ThingTypeUID getThingTypeUID() {
```

Comment:
why does the builder provide the getter operations?

Reply:
This is commented below by Simon so in the next adaption of the PR it will be fixed. Thank you for noticing that.